scripts/common/cmdline_helpers.py

```
@@ -0,0 +1,56 @@
+import os
+import subprocess
+from pathlib import Path
+from tempfile import mkdtemp
+
+from bytecodecompare.prepare_report import FileReport
+from bytecodecompare.prepare_report import parse_cli_output
+
+def enter_tmptest_dir(prefix):
+    """
+        Creates a temporary directory, enters the directory and
+        execute the function inside it.
+        It restores the previous directory after execute the function.
+    """
+    def tmptest_dir_decorator(fn):
+        previous_dir = os.getcwd()
+        def f(*args, **kwargs):
+            try:
+                tmp_dir = mkdtemp(prefix=prefix)
+                os.chdir(tmp_dir)
+                return fn(*args, **kwargs)
+            finally:
+                os.chdir(previous_dir)
+        return f
+    return tmptest_dir_decorator
+
+
+def run_solc(report_file_path: Path, args = None) -> FileReport:
+    """
+        Runs the solidity compiler binary
+        specified by the SOLC environment variable
+    """
+
+    solc_binary_path = os.environ.get("SOLC")
+    if solc_binary_path is None:
+        raise RuntimeError("""
+        `solc` compiler not found.
+        Please ensure you set the SOLC environment variable
+        with the correct path to the compiler's binary.
+        """
+        )
```

I'd rather keep the fact that `cmdlineTests.sh` passes some env variables inside the test itself. Helpers should not make assumptions about that. I think this should be passed to the function via a parameter.

Also, I'd just let the script crash with the default exception if the variable is not defined to keep the test concise.